WIP: Added Swift Example App

NicoleYarroch opened the pull request:
Fixes #620 

This PR is **not ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Smoke tests

### Summary
* Added a Swift example app that mirrors the existing Obj-C example app
* Replace some generic icons with SDL icons
* Reordered example app directory 

### Changelog
##### Breaking Changes
* none

##### Bug Fixes
* [Bug Fix Info]

### Tasks Remaining:
- [ ] Add logging to Swift example app
- [ ] Add vehicle data example?
- [ ] Add a audio pass thru example?

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

joeljfischer commented on SmartDeviceLink_Example/AudioManager.swift:
```diff
@@ -0,0 +1,229 @@
+
+@available(iOS 10.0, *)
+class AudioManager: NSObject {
+    fileprivate let sdlManager: SDLManager!
```
This doesn't need the `!` because it's set in the `init`

joeljfischer commented on SmartDeviceLink_Example/ButtonManager.swift:
```diff
@@ -0,0 +1,159 @@
+    func softButtonImagesVisible() -> SDLSoftButtonObject {
+        let imagesVisibleState = SDLSoftButtonState(stateName: ImagesVisibleSoftButtonImageOnState, text: ImagesVisibleSoftButtonImageOnText, image: nil)
+        let imagesNotVisibleState = SDLSoftButtonState(stateName: ImagesVisibleSoftButtonImageOffState, text: ImagesVisibleSoftButtonImageOffText, image: nil)
+        let softButton: SDLSoftButtonObject? = SDLSoftButtonObject(name: ImagesVisibleSoftButton, states: [imagesVisibleState, imagesNotVisibleState], initialStateName: imagesVisibleState.name) { [unowned self] (buttonPress, buttonEvent) in
```
This init object doesn't create a nullable object. The below guard is unnecessary.

joeljfischer commented on SmartDeviceLink_Example/ButtonManager.swift:
```diff
@@ -0,0 +1,159 @@
+import SmartDeviceLink
+import SmartDeviceLinkSwift
+
+typealias refreshUIHandler = (() -> Void)
```
Types should start with a capital letter

NicoleYarroch replied:
Fixed

joeljfischer commented on SmartDeviceLink_Example/ButtonManager.swift:
```diff
@@ -0,0 +1,159 @@
+    /// Returns a soft button that toggles between two states: on and off. If images are currently visible, the button image toggles; if images aren't visible, the button text toggles.
+    ///
+    /// - Returns: A soft button
+    func softButtonToggle() -> SDLSoftButtonObject {
```
I think we should remove the full image / text toggling soft buttons. It isn't a use case people will be looking to an example for, and it's become very complicated (and will probably confuse people and make SDL appear more complicated than it is). A toggling soft button that changes a soft button when pressed is probably sufficient.

NicoleYarroch replied:
Fixed. Toggle button now switches between two images.

joeljfischer commented on SmartDeviceLink_Example/MenuManager.swift:
```diff
@@ -0,0 +1,193 @@
+// MARK: - Perform Interaction Choice Set Menu
+
+private extension MenuManager {
+    static let choiceSetId = 100
```
Constants should start with a capital letter

NicoleYarroch replied:
Fixed in both Swift and Obj-C Ex. apps

joeljfischer commented on SmartDeviceLink_Example/MenuManager.swift:
```diff
@@ -0,0 +1,193 @@
+    }
+}
+
+// MARK: - Perform Interaction Choice Set Menu
```
I think the perform interaction menu should be a separate class / file

NicoleYarroch replied:
Fixed. Added PICS menu manager to both Obj-C and Swift Ex. apps

joeljfischer commented on SmartDeviceLink_Example/ProxyManager.swift:
```diff
@@ -0,0 +1,278 @@
+        case .full:                // The SDL app is in the foreground
+            // Always try to show the initial state to guard against some possible weird states. Duplicates will be ignored by Core.
+            showInitialData()
+        case .limited: break        // The SDL app's menu is open
```
Limited is when an active NAV / MEDIA app is in the background (and may still be controlled even though its UI isn't onscreen).

NicoleYarroch replied:
Fix in both Swift and Obj-C Ex. apps

joeljfischer commented on SmartDeviceLink_Example/MenuManager.swift:
```diff
@@ -0,0 +1,193 @@
+    ///
+    /// - Returns: A SDLPerformInteraction request
+    class func createPerformInteraction() -> SDLPerformInteraction {
+        let performInteraction = SDLPerformInteraction(initialPrompt: PICSInitialPrompt, initialText: PICSInitialText, interactionChoiceSetIDList: [choiceSetId as NSNumber], helpPrompt: PICSHelpPrompt, timeoutPrompt: PICSTimeoutPrompt, interactionMode: .both, timeout: 10000)
```
The `interactionMode` should come from the `triggerSource` of the menu selection. If touched -> Manual only, voice -> voice only.

NicoleYarroch replied:
Fixed in both Swift and Obj-C Ex. apps.

joeljfischer commented on SmartDeviceLink_Example/MenuManager.swift:
```diff
@@ -0,0 +1,193 @@
+import SmartDeviceLink
+import SmartDeviceLinkSwift
+
+class MenuManager: NSObject {
```
Why doesn't this exist in the Obj-C app?

NicoleYarroch replied:
Fixed. Now exists.

NicoleYarroch commented:
> Also, the Product Name of the swift target should be updated.

Fixed.

> The Obj-C odometer line says "Unsubscribed" while the Swift one says "Disallowed"

Fixed.

joeljfischer commented on SmartDeviceLink_Example/Enum+SDLConnectionType.swift:
```diff
@@ -0,0 +1,14 @@
+
+import Foundation
+
+enum SDLConnectionType {
```
This can just be in the proxy class, they should all start lower case (tcp / iap is fine)
